pypi_server/handlers/pypi/package.py

```diff
@@ -112,7 +111,7 @@ def find_file(self, package, version, filename):
     @classmethod
     @coroutine
     def fetch_remote_file(cls, pkg_file):
-        response = yield cls.HTTP_CLIENT.fetch(pkg_file.url)
+        response = yield AsyncHTTPClient().fetch(pkg_file.url)
```

Comment:
AsyncHTTPClient() actually always returns the same instance of the configured class, not a new one, unless force_instance is set to True.

Because the call to AsyncHTTPClient() is done during the class definition, any subsequent calls to AsyncHTTPClient.configure() become useless, as there was already an instance created before. This is a prerequisite for the HTTP proxy support to work.